src/Components/Pages/SetWorkLoadPage.js

```diff
@@ -39,52 +44,37 @@ const SetWorkloadPage = ({
 
   useEffect(() => {
     if (data && workload.includes('Extended')) {
-      setExtendedReleaseRepositories(
-        data
-          .find((product) => extendedReleaseProduct == product.name)
-          .configurations.find(
-            (configuration) => extendedReleaseVersion == configuration.version
-          ).repositories
+      const selectedProduct = data.find(
+        (product) => extendedReleaseProduct === product.name
       );
+      const selectedVersion = selectedProduct?.configurations.find(
+        (configuration) => extendedReleaseVersion === configuration.version
+      );
+      setExtendedReleaseRepositories(selectedVersion?.repositories || []);
     } else {
       setExtendedReleaseRepositories([]);
     }
   }, [data, extendedReleaseProduct, extendedReleaseVersion]);
-
   return (
     <>
       <Title headingLevel="h2" className="pf-v5-u-mb-sm">
-        Select Workload
+        {isEditMode ? 'Edit Workload' : 'Select Workload'}
```

Comment:
Suggestion: Same here, consider if the slight difference in wording is worth depending on the context of the modal. If not, we can remove that dependence here too.

Reply:
For the naming conventions changes I was trying to match the mockups , I do understand the slight differences might not be necessary especially with the conditional check spam. I would be ok with leaving it to "select workload" but i can double check with UX if this is acceptable?

Reply:
Yeah, matching the mockups is great and the right hunch. I would ask UX if its okay just cause it will reduce component coupling a lot, but if they have a strong preference for the edit-specific wording that's good too!